### PR TITLE
adjust function select_elec_network

### PR DESCRIPTION
**Pull request by CarlosEpia**
Fix a small bug when running market optimization while the parameter "cluster_foreign_AC" = True.
#719

### CODE REVIEW

File: etrago/cluster/electrical.py [hunk 1]
```diff
@@ -645,14 +648,20 @@ def unify_foreign_buses(etrago):
...
     Preprocesses an Etrago object to prepare it for network clustering.
 
     Parameters
     ----------
     etrago : Etrago
         An instance of the Etrago class
+    used_for : string
+        provide information about the objective of the preprocessing. Which
+        process is going to use the result. e.g. "spatial_clustering",
+        "market_model".
```
Comment (ClaraBuettner) [inline etrago/cluster/electrical.py:664]: This parameter seems to be very similar to the other new parameter "apply_on" that is already in `dev`. 
Is the "used_for" parameter always "market_model" when applied on the market model and "spatial_clustering" if it is applied on the grid_model /etrago.network?

Comment (CarlosEpia) [reply etrago/cluster/electrical.py:664]: Yes, it is. I will adjust my PR to work using the already existing parameter "apply_on".

**Comment by CarlosEpia** [comment]
Ready for a new check.

File: etrago/cluster/electrical.py [hunk 2]
```diff
@@ -477,6 +477,12 @@ def select_elec_network(etrago, apply_on="grid_model"):
     ----------
     etrago : Etrago
         An instance of the Etrago class
+    apply_on: str
+        gives information about the objective of the output network. If
+        "spatial_clustering" is provided, the value assigned in the args for
```
Comment (ClaraBuettner) [inline etrago/cluster/electrical.py:482]: Is it correct that you can choose 'spatial_clustering' here? 
The code later on only allows 'grid_model' or 'market_model'

File: etrago/cluster/electrical.py [hunk 3]
```diff
@@ -496,7 +502,13 @@ def select_elec_network(etrago, apply_on="grid_model"):
             """
         )
     settings = etrago.args["network_clustering"]
-    if settings["cluster_foreign_AC"]:
+
+    if apply_on == "grid_model":
+        include_foreign = settings["cluster_foreign_AC"]
+    elif apply_on == "market_model":
+        include_foreign = True
```
Comment (ClaraBuettner) [inline etrago/cluster/electrical.py:509]: There is no else statement

**Review by ClaraBuettner — APPROVED** [review]

**Pull request merged**